### PR TITLE
contrib: Add a standalone installer creation script

### PR DESCRIPTION
**Pull request by superm1**
This is meant to produce standalone installer binaries that can operate
on machines without current tools or internet access.

It works by wrapping around the container technologies snap and flatpak
and putting all the pieces together.

### CODE REVIEW

File: contrib/standalone-installer/make.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+def download_snap (directory, channel):
+    cmd = ['snap', 'download', 'fwupd']
+    if channel is not None:
+        cmd += ['--channel', channel]
+    if 'DEBUG' in os.environ:
+        print(cmd)
+    subprocess.run (cmd, cwd=directory)
+    for f in os.listdir (directory):
+        if f.endswith(".assert"):
```
Comment (hughsie) [inline contrib/standalone-installer/make.py]: assert?

File: contrib/standalone-installer/assets/header [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#!/bin/sh -e
```
Comment (hughsie) [inline contrib/standalone-installer/assets/header]: Any reason why not python3? I kinda hate bash scripts.

**Comment by superm1** [comment]
OK, I've reworked it as fully python3 now.

**Comment by hughsie** [comment]
Lgtm, thanks for doing that.

**Comment by superm1** [comment]
> Lgtm, thanks for doing that.

Sure.  Can you please update your flatpak that it is pointing at?  I notice it's a bit out of date.  Also, any updates on moving it to flathub?  (Of course this will have to change when you do)

**Comment by hughsie** [comment]
> Also, any updates on moving it to flathub

https://github.com/flathub/flathub/pull/595 -- fingers crossed we can get it in ASAP.

**Pull request merged**